core/common/src/main/java/alluxio/security/authorization/AccessControlList.java

```diff
@@ -528,6 +560,8 @@ public TAcl toThrift() {
   }
 
   /**
+   * Converts a list of string entries into an AccessControlList or a DefaultAccessControlList.
+   * It assumes the stringEntries contain all default entries or normal entries.
```

Comment:
I think this assumption makes the logic in this class tricky (other methods too). Would it be simpler if `DefaultAccessControlList` is removed, and this class just had a member `List<AclEntry> mDefaultEntries = new ArrayList();`, and we add a few methods (similar to the methods in the current `DefaultAccessControlList`? What are the downsides to that approach?

Reply:
The downside is that we would be duplicating a lot of the logic to `removeEntry` and `setEntry`.  Would it be better if we actually check if all the string entries are consistently `default` or not ? and throw an exception if they are inconsistent with each other?

Reply:
Ok. We can keep it like this. We may revisit this later.